app/services/custom_landing_page/link_resolver.rb

```diff
@@ -112,5 +112,66 @@ def categories
         @_categories ||= Category.where(community_id: @_cid).to_a
       end
     end
+
+    class ListingResolver
+      def initialize(cid, locale, name_display_type)
+        @_cid = cid
+        @_locale = locale
+        @_name_display_type = name_display_type
+      end
+
+      def call(type, id, _)
+        listing = Listing.includes(:author).find_by(community_id: @_cid, id: id)
```

Comment:
We have stores for accessing marketplace data and for accessing landing page data. Following the same pattern listings and categories should have landing page service specific minimal stores too to separate resolvers from AR objects.